### PR TITLE
Custom filename entry

### PR DESCRIPTION
**Pull request by paulirish**
#### Hey there tracing user! 
***Are you tired of the same old `trace (2).json` ?***
image

***Wouldn't you like more meaningful names?***
image

***I've got just the patch for you!***

<hr>

I took a shot at more meaningful filenames. 

image

We use a lovely blocking `prompt()` call to ask for a more meaningful name. If you ignore it and hit enter, you'll get a decent looking timestamp. And if you type, well… you'll get something better looking.

<hr>

I explored two options aside from the `prompt()` approach.

## auto-magical relevant name
I tried to pull a more meaningful name out of the process thread `userFriendlyName`s but it seems unreliable at the moment. 

## happy names
Gfycat uses unique word combinations for their urls, like http://gfycat.com/MagnificentHighCapybara and https://gfycat.com/TameRepentantAfricanbushviper  [[sources](http://www.reddit.com/r/gfycat/comments/2bkl5j/gfycat_url_generator/)]. We could do something similar.

In the end settled on prompt figuring it's the benefits of no more `trace (2).json` outweigh the costs of `prompt()`. 

What does everyone think?

### CODE REVIEW

**Comment by natduca** [comment]
omgahhhhh lgtm (and made you committer so you can merge yourself when nit is fixed)

**Comment by paulirish** [comment]
Thx! Merging.

Very happy to revert this back if people get upset with the prompting, let's wait and see.

**Comment by dj2** [comment]
heh, https://github.com/dj2/Moniker

File: trace_viewer/extras/about_tracing/profiling_view.html [hunk 1]
```diff
@@ -402,6 +402,34 @@
...
+      var reservedRe = /^\.+$/;
+
+      var filename;
+      var defaultName = this.activeTrace_.filename;
+      var custom = prompt('Filename? (.json appended) Or leave blank:');
+      if (custom === null)
+        return false;
+
+      if (custom) {
+        filename = defaultName.replace(/\.json$/, ' ' + custom) + '.json';
```
Comment (dj2) [inline trace_viewer/extras/about_tracing/profiling_view.html]: Doesn't this mean the name will always start with a blank character?

**Comment by paulirish** [comment]
@dj2 omg. that's great. :) yes exactly.

File: trace_viewer/extras/about_tracing/profiling_view.html (hunk 1 above)
Comment (paulirish) [inline trace_viewer/extras/about_tracing/profiling_view.html]: I'm keeping the "trace" of `this.activeTrace_.filename` intact and adding a space after it.

**Pull request merged**